src/Symfony/Component/DependencyInjection/Tests/ContainerBuilderTest.php

```diff
@@ -548,6 +548,19 @@ public function testResolveEnvValues()
         unset($_ENV['DUMMY_ENV_VAR']);
     }
 
+    public function testCompileWithResolveEnv()
+    {
+        $_ENV['DUMMY_ENV_VAR'] = 'du%%y';
+
+        $container = new ContainerBuilder();
+        $container->setParameter('bar', '%% %env(DUMMY_ENV_VAR)%');
```

Comment:
What will happen in the other two cases: 1) undefined env var; 2) undefined env var, but default value defined for it. Should we add those cases to this test too?